apache/airflow
# Exponential Backoff Not Functioning in BaseSensorOperator Reschedule Mode

### Fix: Increment try_number/poke_count in BaseSensorOperator for correct exponential backoff in reschedule mode.

What happened?
- The `BaseSensorOperator` does not correctly implement exponential backoff when in `reschedule` mode.
- The sensor reschedules at a constant interval instead of increasing the interval exponentially.

What you think should happen instead?
- The `try_number` (or `poke_count`) should increment on each reschedule too, ensuring the backoff interval increases exponentially.

How to reproduce?
- Set up a `BaseSensorOperator` with `mode="reschedule"` and `exponential_backoff=True`.
- Observe the interval between reschedules; it remains constant instead of increasing exponentially.


#6654 









---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).
